drivers/neo4j-embedded/src/main/java/org/commonjava/maven/atlas/graph/spi/neo4j/io/Conversions.java

```diff
@@ -232,12 +233,34 @@ public static int countArrayElements( final String property, final PropertyConta
         return refs;
     }
 
-    public static List<ProjectRelationship<?, ?>> convertToDetachedRelationships( final Iterable<Relationship> relationships )
+    /**
+     * Converts an iterable of relationships to project relationships detached from database.
+     *
+     * @param relationships iterable of {@link AbstractNeoProjectRelationship} or {@link Relationship}
+     * @return list of detached relationships
+     * @throws IllegalArgumentException if an iterable of unsupported classes is passed
+     */
+    public static List<ProjectRelationship<?, ?>> convertToDetachedRelationships( final Iterable<?> relationships )
     {
         final List<ProjectRelationship<?, ?>> rels = new ArrayList<ProjectRelationship<?, ?>>();
-        for ( final Relationship relationship : relationships )
+        Iterator<?> iterator = relationships.iterator();
+        while ( iterator.hasNext() )
         {
-            final AbstractNeoProjectRelationship<?, ?, ?> rel = Conversions.toProjectRelationship( relationship );
+            final AbstractNeoProjectRelationship<?, ?, ?> rel;
+            Object next = iterator.next();
+            if ( next instanceof AbstractNeoProjectRelationship )
+            {
+                rel = ( AbstractNeoProjectRelationship<?, ?, ?>) next;
+            }
+            else if ( next instanceof AbstractNeoProjectRelationship )
+            {
+                rel = Conversions.toProjectRelationship( ( Relationship ) next );
+            }
```

Comment:
Is this first else-if dead code? It seems to have the same condition as the initial if() statement.